src/common/transformations/src/transformations/low_precision/mark_dequantization_subgraph.cpp

```diff
@@ -5,114 +5,174 @@
+        // Multiply and Subtract have to be marked as dq
+        set_rt_info(pt_map, mark_as_dequantization_node, {subtract_pattern, multiply_pattern}, {/* not applicable */});
+
+        // Convert might be presented on scales, zp and data_input.
+        // Depending on the transformation arguments they have to be marked/unmarked with disable_cf rt_info.
+        NodeVector converts_to_mark = {convert_pattern};
+        NodeVector converts_to_unmark = {};
+
+        if (fold_subtract_const) {
+            converts_to_unmark.push_back(zp_convert_pattern);
+        } else {
+            converts_to_mark.push_back(zp_convert_pattern);
         }
 
-        if (subtract_it != pattern_map.end()) {
-            // mark Subtract as dequantization node
-            ov::mark_as_dequantization_node(subtract_it->second.get_node_shared_ptr());
-            auto zero_point = pattern_map.at(zero_point_pattern).get_node_shared_ptr();
-            if (ov::is_type<ov::op::v0::Convert>(zero_point) &&
-                input_precision == zero_point->get_input_element_type(0) &&
-                ov::is_type<ov::op::v0::Constant>(zero_point->get_input_node_ptr(0))) {
-                if (!fold_subtract_const) {
-                    // disable ConstantFolding also for Convert on zero_point
-                    // so we don't have to constantfold it and then convert it back to
-                    // low precision in LP transformations
-                    ov::disable_constant_folding(zero_point);
-                    ov::enable_keep_const_precision(zero_point->get_input_node_shared_ptr(0));
-                } else {
-                    ov::enable_constant_folding(zero_point);
-                    ov::disable_keep_const_precision(zero_point->get_input_node_shared_ptr(0));
-                }
-            }
+        if (fold_multiply_const) {
+            converts_to_unmark.push_back(scale_convert_pattern);
+        } else {
+            converts_to_mark.push_back(scale_convert_pattern);
         }
 
-        // mark Multiply as dequantization node
-        ov::mark_as_dequantization_node(multiply);
-        auto scale = multiply->get_input_node_shared_ptr(1);
-        if (ov::is_type<ov::op::v0::Convert>(scale) &&
-            ov::is_type<ov::op::v0::Constant>(scale->get_input_node_ptr(0))) {
-            if (disable_fold_multiply_const) {
-                ov::disable_constant_folding(scale);
-                ov::unmark_as_decompression(scale);
-                ov::enable_keep_const_precision(scale->get_input_node_shared_ptr(0));
-            }
+        set_rt_info(pt_map, disable_constant_folding, converts_to_mark, precisions);
+        set_rt_info(pt_map, enable_constant_folding, converts_to_unmark, precisions);
+
+        // Move Reshape/Unsqueeze ops up to fold them in ConstantFolding.
+        auto changed = swap_nodes(pt_map, zp_convert_pattern, zp_reshape_pattern);
+        changed = changed || swap_nodes(pt_map, scale_convert_pattern, scale_reshape_pattern);
```

Comment:
swap_nodes() won't be called if changed is already true.

Reply:
fixed